ssm/ssm2.py

```diff
@@ -144,27 +144,32 @@ def on_message(self, headers, body):
             empaid = 'noid'
 
         log.info("Received message. ID = %s", empaid)
-        raw_msg, signer = self._handle_msg(body)
+        raw_msg, signer, error = self._handle_msg(body)
 
         try:
-            if raw_msg is None:  # the message has been rejected
+            if raw_msg is None or error is not None:
+            # the message has been rejected
+
                 if signer is None:  # crypto failed
-                    err_msg = 'Could not extract message.'
                     signer = 'Not available.'
-                else:  # crypto ok but signer not verified
-                    err_msg = 'Signer not in valid DNs list.'
-                log.warn("Message rejected: %s", err_msg)
+
+                log.warn("Message rejected: %s", error)
```

Comment:
I want to say it was because pylint didn't like the variable name, but it seems happy enough now so I'm not sure.

happy to change it back?

Reply:
I suppose I'd prefer to avoid unnecessary changes to variable names.

Just change the ones in `on_message` back. The one's in `_handle_msg` can stay as they are as they're new.

Reply:
I agree